userspace/libsinsp/sinsp_suppress.cpp

```diff
@@ -145,3 +149,37 @@ bool libsinsp::sinsp_suppress::is_suppressed_tid(uint64_t tid) const {
 	}
 	return m_suppressed_tids.find(tid) != m_suppressed_tids.end();
 }
+void libsinsp::sinsp_suppress::initialize() {
+	m_tids_tree.clear();
+}
+
+void libsinsp::sinsp_suppress::handle_thread(uint64_t tid,
+                                             uint64_t parent_tid,
+                                             const std::string &comm) {
+	if(tid == 0) {
+		return;
+	}
+
+	// Add the comm to tid
+	m_tids_tree[tid].m_comm = comm;
+	m_tids_tree[tid].m_tid = tid;
+
+	// Add child to parent
+	m_tids_tree[parent_tid].m_children.push_back(tid);
+}
+
+void libsinsp::sinsp_suppress::finalize() {
+	// We generate the suppressed tids from the tid tree.
+	// The tree is built during the /proc scan, so we can
+	// use it to find all the children of a given tid.
+
+	for(auto it = m_tids_tree.begin(); it != m_tids_tree.end(); it++) {
+		auto &[_, node] = *it;
+
+		if(is_suppressed_tid(node.m_tid)) {
+			for(auto child_tid : node.m_children) {
+				suppress_tid(child_tid);
+			}
+		}
+	}
+}
```

Comment:
We need to clear `m_tids_tree` after `finalize` ends to spare memory. @deepskyblue86 what do you think? maybe also, the start refresh callback, if we clear here, is no longer needed (Or we can leave it for a defensive approach and for consistency as well)

Reply:
makes sense